src/components/course/data/hooks.jsx

```diff
@@ -789,13 +792,11 @@ export const useUserSubsidyApplicableToCourse = ({
     if (applicableUserSubsidy) {
       setUserSubsidyApplicableToCourse(applicableUserSubsidy);
       setMissingUserSubsidyReason(undefined);
-    } else {
-      // If have not yet determined whether there is an applicable subsidy access policy, fallback
-      // to checking against legacy subsidies.
+    } else if (subscriptionLicense && containsContentItems) {
```

Comment:
If we want subscriptions shown, evaluated, whatever first; this `else if` block needs to actually lead this if statement starting on line 792.